R/smtp_send.R

```diff
@@ -92,73 +119,99 @@ smtp_send <- function(email,
+         " * `creds()`: allows for manual specification of SMTP credentials",
+         call. = FALSE)
+  }
+
+  # If whatever is provided to `credentials` does not have a
+  # `blastula_creds` class, determine whether that value is a
+  # single-length character vector (which is upgraded through
+  # the `creds_file()` function); if it's anything else, stop
+  # the function with a message
+  if (!inherits(credentials, "blastula_creds")) {
+    if (is.character(credentials) && length(credentials) == 1) {
+
+      credentials <- creds_file(credentials)
+
+    } else {
+      stop("The value for `credentials` must be a `blastula_creds` object\n",
+           "* see the article in `?creds` for information on this",
+           call. = FALSE)
+    }
+  }
+
+  # Create the `creds_list` object by processing whichever
+  # credentials type (key, file, or manual) was specified
+  # in `credentials`
+  if (inherits(credentials, "creds_key")) {
+    creds_list <- get_smtp_keyring_creds(key_name = credentials$key_name)
+  } else if (inherits(credentials, "creds_file")) {
+    creds_list <- get_smtp_file_creds(file_name = credentials$file)
+  } else if (inherits(credentials, "creds")) {
+    creds_list <- credentials %>% unclass()
+  }
+
   # Write the inlined HTML message out to a file
-  email$html_str %>%
-    writeLines(con = "message_inlined.html")
+  email$html_str %>% writeLines(con = "message_inlined.html")
 
   # Remove the file after the function exits
   on.exit(file.remove("message_inlined.html"))
 
   # Handle a subject line that's not provided and use
   # `glue::glue()` for customizing a given `subject`
-  if (is.null(subject)) {
-    subject <- "<no subject>"
-  } else {
-    subject <- glue::glue(subject) %>% as.character()
-  }
+  subject <-
+    glue::glue(subject) %>%
+    as.character() %||%
+    "<no subject>"
 
   # Create comma-separated addresses for
   # `to`, `cc`, and `bcc`
   to <- make_address_list(to)
   cc <- make_address_list(cc)
   bcc <- make_address_list(bcc)
 
-  # If a path to a credentials file is provided,
-  # read in the values
-  if (!is.null(creds_file)) {
-
-    credentials <- readRDS(creds_file) %>% as.list()
-
-    # Overwrite sender name if one is provided
-    if (!is.null(sender)) credentials$sender <- sender
-
-  } else {
-
-    if (is.null(password)) {
-      password <- getPass::getPass("Enter the SMTP server password: ")
-    }
-
-    credentials <-
-      list(
-        sender = sender,
-        host = host,
-        port = port,
-        user = user,
-        password = password,
-        use_ssl = use_ssl,
-        authenticate = authenticate
-      )
-  }
-
   # Set the `ssl` flag depending on the options provided
-  if (credentials$use_ssl) {
+  if (creds_list$use_ssl) {
     ssl_opt <- no_options()
   } else {
     ssl_opt <- no_arg()
   }
 
+  # Set the `sender_name` to `no_arg()` if not provided
+  sender_name_opt <- creds_list$sender_name %||% no_arg()
```

Comment:
Instead of defaulting to `no_arg()`, just throw an error

Reply:
I was wrong here. Turns out the sender name isn't strictly required by `mailsend-go` (I've been successfully sending test emails without that field populated).